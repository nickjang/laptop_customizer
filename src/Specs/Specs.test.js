import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Specs from './Specs';

describe('<Specs />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Specs />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(<Specs />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders feature with options when props are supplied', () => {
    let feature = [
      {
        name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
        cost: 1500
      },
      {
        name: '15',
        cost: 100
      }
    ];
    
    const wrapper = shallow(
      <Specs key="feature-1" idx={1} featureName="Display"
        feature={feature} updateFeature={() => { }} selectedFeatureName="15" />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});