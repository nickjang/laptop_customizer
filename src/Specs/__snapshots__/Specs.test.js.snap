// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Specs /> renders empty when no props supplied 1`] = `
<fieldset
  className="feature"
  key="-0"
>
  <legend
    className="feature__name"
  >
    <h3 />
  </legend>
</fieldset>
`;

exports[`<Specs /> renders feature with options when props are supplied 1`] = `
<fieldset
  className="feature"
  key="Display-1"
>
  <legend
    className="feature__name"
  >
    <h3>
      Display
    </h3>
  </legend>
  <Parts
    feature="Display"
    item={
      Object {
        "cost": 1500,
        "name": "15.6\\" UHD (3840 x 2160) 60Hz Bright Lights and Knobs",
      }
    }
    itemHash="\\"name\\":\\"15.6\\"-UHD-(3840-x-2160)-60Hz-Bright-Lights-and-Knobs\\"\\"cost\\":1500"
    key="\\"name\\":\\"15.6\\"-UHD-(3840-x-2160)-60Hz-Bright-Lights-and-Knobs\\"\\"cost\\":1500"
    selectedFeatureName="15"
    updateFeature={[Function]}
  />
  <Parts
    feature="Display"
    item={
      Object {
        "cost": 100,
        "name": "15",
      }
    }
    itemHash="\\"name\\":\\"15\\"\\"cost\\":100"
    key="\\"name\\":\\"15\\"\\"cost\\":100"
    selectedFeatureName="15"
    updateFeature={[Function]}
  />
</fieldset>
`;
